front/src/modules/simulationResult/components/SpaceTimeChart/WaypointsPanel.tsx

```diff
@@ -0,0 +1,194 @@
+
+  const allWaypointsSelected = useMemo(
+    () => selectedWaypoints.size === waypoints.length,
+    [selectedWaypoints, waypoints]
+  );
+
+  const isIndeterminate = useMemo(
+    () => selectedWaypoints.size > 0 && !allWaypointsSelected,
+    [selectedWaypoints, allWaypointsSelected]
+  );
+
+  const openModal = () => {
+    modalRef.current?.showModal();
+    const filteredWaypointsIds = filteredWaypoints.map((waypoint) => waypoint.id);
+    const filteredWaypointsIndexes = waypoints
+      .map((waypoint, index) => (filteredWaypointsIds.includes(waypoint.id) ? index : -1))
+      .filter((index) => index !== -1);
+
+    setSelectedWaypoints(new Set(filteredWaypointsIndexes));
+  };
+
+  const closeModal = () => {
+    modalRef.current?.close();
+    setWaypointsPanelIsOpen(false);
+  };
+
+  useModalFocusTrap(modalRef, closeModal);
+
+  const handleSubmit = () => {
+    if (selectedWaypoints.size < 2) {
+      setIsNotEnoughSelectedWaypoints(true);
+      return;
+    }
+    const newFilteredWaypoints = waypoints.filter((_, i) => selectedWaypoints.has(i));
+    setFilteredWaypoints(newFilteredWaypoints);
+
+    // We need to removed the id because it can change for waypoints added by map click
+    const simplifiedPath = projectionPath.map((waypoint) => omit(waypoint, ['id', 'deleted']));
+
+    // TODO : when switching to the manchette back-end manager, remove all logic using
+    // cleanScenarioLocalStorage from projet/study/scenario components (single/multi select)
+    localStorage.setItem(
+      `${timetableId}-${JSON.stringify(simplifiedPath)}`,
+      JSON.stringify(newFilteredWaypoints)
+    );
+
+    closeModal();
+  };
+
+  useEffect(() => {
+    if (waypointsPanelIsOpen) {
+      openModal();
+    }
+  }, [waypointsPanelIsOpen]);
+
+  // TODO : fix checkbox component in osrd-ui to handle shift + click because of a firefox bug (#510)
+  const handleWaypointClick = (event: React.MouseEvent<HTMLInputElement>, index: number) => {
+    event.stopPropagation();
+    if (event.shiftKey && lastSelectedIndex !== undefined) {
+      const minIndex = Math.min(index, lastSelectedIndex);
+      const maxIndex = Math.max(index, lastSelectedIndex);
+      const newSet = new Set(selectedWaypoints);
+
+      // If the lastSelectedIndex has been checked, we check all waypoints of the range
+      // If it has been unchecked, we uncheck all waypoints of the range
+      if (newSet.has(lastSelectedIndex)) {
+        for (let i = minIndex; i <= maxIndex; i += 1) {
+          newSet.add(i);
+        }
+      } else {
+        for (let i = minIndex; i <= maxIndex; i += 1) {
+          newSet.delete(i);
+        }
+      }
+      setSelectedWaypoints(newSet);
+    } else {
+      setSelectedWaypoints((prev) => {
+        const newSet = new Set(prev);
+        if (newSet.has(index)) {
+          newSet.delete(index);
+        } else {
+          newSet.add(index);
+        }
+        return newSet;
```

Comment:
You don't need the prev here I think :)
```suggestion
        const newSet = new Set(selectedWaypoints); // to move l97
        if (newSet.has(index)) {
          newSet.delete(index);
        } else {
          newSet.add(index);
        }
        setSelectedWaypoints(newSet) // this can even be moved outside the if ... else ...
```

Reply:
Done. Because of eslint, had to change a bit the last else.